cmd/gateway/commands.go

```diff
@@ -209,6 +209,11 @@ func createControllerCommand() *cobra.Command {
 				imageSource = "unknown"
 			}
 
+			buildOs := os.Getenv("BUILD_OS")
```

Comment:
Couldn't you just call this directly in the collector, rather than calling here and passing the var all the way through the code?